doc/js/scripts/algolia-searchbox.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * This script is used initialize Algolia DocSearch on each page. It will hydrate the
+ * container with ID `docsearch` (see `doc/templates/algolia-searchbox.html`) with the
+ * Algolia search widget.
+ */
+
+document.addEventListener("DOMContentLoaded", () => {
+  // Figure out how to route to the search page from the current page where we will show
+  // all search results
+  const pagename = DOCUMENTATION_OPTIONS.pagename;
+  let searchPageHref = "./";
+  for (let i = 0; i < pagename.split("/").length - 1; i++) {
+    searchPageHref += "../";
+  }
+  searchPageHref += "algolia-search.html";
+
+  // Function to navigate to the all results page
+  const navigateToResultsPage = () => {
+    const link = document.getElementById("sk-search-all-results-link");
+    if (link !== null) {
+      // If there is the "see all results" link, just click it
+      link.click();
+      return;
+    }
+
+    const inputBox = document.getElementById("docsearch-input");
+    if (inputBox === null || inputBox.value === "") {
+      // If we cannot get the input box or the input box is empty, navigate to the
+      // all results page with no query
+      window.location.assign(searchPageHref);
+      return;
+    }
+    // Navigate to the all results page with query constructed from the input
+    const query = new URLSearchParams({ q: inputBox.value });
+    window.location.assign(`${searchPageHref}?${query}`);
+  };
+
+  // Initialize the Algolia DocSearch widget
+  docsearch({
+    container: "#docsearch",
+    appId: SKLEARN_ALGOLIA_APP_ID,
+    apiKey: SKLEARN_ALGOLIA_API_KEY,
+    indexName: SKLEARN_ALGOLIA_INDEX_NAME,
+    placeholder: "Search the docs ...",
+    searchParameters: { attributesToHighlight: ["hierarchy.lvl0"] },
```

Comment:
Apparently there is a bug that explain why I could not get the desired rendering in the instant box:
https://github.com/algolia/docsearch/issues/2294